_posts/2018-09-18-How-to-build-a-Serverless-Application-with-AWS-Lambda-and-DynamoDB.md

```diff
@@ -0,0 +1,563 @@
+
+const AWS = require('aws-sdk');
+const docClient = new AWS.DynamoDB.DocumentClient({region: 'eu-west-1'});
+
+exports.handler = function(event, context, callback){
+    console.log('processing event: %j', event);
+
+    let scanningParameters = {
+        TableName: 'CodingTips',
+        Limit: 100 //maximum result of 100 items
+    };
+
+    //In dynamoDB scan looks through your entire table and fetches all data
+    docClient.scan(scanningParameters, function(err,data){
+        if(err){
+            callback(err, null);
+        }else{
+            callback(null,data);
+        }
+    });
+}
+```
+
+**Save** the Lambda Function to persist the changes.  
+
+* The handler function is the function where the Lambda execution starts when the Lambda is triggered.
+* The event parameter contains the data from the event that triggered the function.
+* The `scanningParameters` are used to configure the scan of the table.
+* This function scans the DynamoDB table for the first 100 items it finds.
+* `docClient.scan(scanningParameters, function(err,data)` executes the scan and returns either the result or the error that occurred.
+
+## Test write-Lambda
+All right! Let's test this thing..
+On the Lambda Function configuration page you see a dropdown and test button in the upper right corner.
+Click the dropdown and configure a new test event.
+I called mine 'Test' and added an empty test event.
+
+<div style="text-align: center;">
+  <img src="/img/2018-09-18-How-to-Build-a-Serverless-Application/Lambda-CodingTips_Scan-Created_Empty_Test_Event.png" width="60%" height="60%">
+</div>  
+
+**Save** it and you are ready to test the Lambda.  
+From the dropdown select your test event and hit the **Test** button!
+Nice one, this returns the items in your table:
+
+<div style="text-align: center;">
+  <img src="/img/2018-09-18-How-to-Build-a-Serverless-Application/Lambda-CodingTips_Scan-Execution_Result_Succeeded.png" width="100%">
+</div>
+
+# API Gateway: Access the scan Lambda
+Mmmh, fine.. We can trigger the Lambda Function with a test event.
+But we want to be able to trigger it from anywhere using a URL.
+In the designer view of the lambda you can still see **add triggers from the left**.
+Well, let's add that trigger!
+To expose a Lambda Function AWS provides the **API Gateway**.
+
+Under **Services** navigate to **API Gateway**. 
+> Amazon API Gateway is a fully managed service that makes it easy for developers to create, publish, maintain, monitor, and secure APIs at any scale. 
+With a few clicks in the AWS Management Console, you can create an API that acts as a “front door” for applications -- from AWS docs: <a target="_blank" href="https://aws.amazon.com/api-gateway/">https://aws.amazon.com/api-gateway</a>
+
+Basically this is the Service you use to create all of your API's.  
+* Click **Create api** and name your api **CodingTips**
+* Add a description if you like
+* Leave the **Endpoint Type** to regional and **create API**
```

Comment:
Create (capital 'C')

Reply:
I will set the text to "Create API" since this is the text shown on the button in the AWS console